src/lib.rs

```diff
@@ -28,15 +28,18 @@ pub enum Eszip {
 }
 
 /// This future needs to polled to parse the eszip file.
-type EszipParserFuture<R> =
-  Pin<Box<dyn Future<Output = Result<futures::io::BufReader<R>, ParseError>>>>;
+type EszipParserFuture<R> = Pin<
+  Box<
+    dyn Future<Output = Result<futures::io::BufReader<R>, ParseError>> + Send,
+  >,
+>;
```

Comment:
Note for reviewers: here `Send` bound is added. This is not directly related to implementing `IntoIterator`, but this should be helpful when someone parses an eszip archive using `Eszip::parse` method and wants to drive the returned `EszipParserFuture` using `tokio::spawn` or something similar, because `tokio::spawn` requires the passed-in future to be `Send`.

Reply:
Actually wasm binding doesn't work well with Send bound. Removed it

Reply:
On third thought, without `Send` it would be really difficult to work with in the caller's side. I created a new method `Eszip::parse_local` that accepts `!Send` reader and returns a future that is `!Send`. And `Eszip::parse` takes `Send` reader.
Callers can choose whichever suits their needs.